src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,34 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new IncorrectUserDataException("User is null");
+        }
+        if (user.getLogin() == null) {
+            throw new IncorrectUserDataException("User login is null");
+        }
+        if (user.getAge() == null) {
+            throw new IncorrectUserDataException("User age is null");
+        }
+        if (user.getAge() < 0 || user.getAge() > 100) {
```

Comment:
```suggestion
        if (user.getAge() > 100) {
```
redunant check. there is already a check in case the user is under 18